tests/metrics/test_remap_instance_id.py

```
@@ -0,0 +1,88 @@
+    ["single_instance", [[0, 0], [3, 3]], True, [[0, 0], [1, 1]]],
+]
+
+
+def _reference_remap(pred: torch.Tensor, by_size: bool = False) -> torch.Tensor:
+    """The original per-instance-loop implementation, kept as the behavioral reference."""
+    pred_id = [i for i in pred.unique() if i != 0]
+    if not pred_id:
+        return pred
+    if by_size:
+        instance_size = [(pred == instance_id).sum() for instance_id in pred_id]
+        pair_list = sorted(zip(pred_id, instance_size), key=lambda x: x[1], reverse=True)
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)